noir-projects/aztec-nr/aztec/src/utils/u64s_packing.nr

```diff
@@ -0,0 +1,88 @@
+global U64_SHIFT_MULTIPLIER: Field = 2.pow_32(64);
+
+pub fn pack_two_u64s(high: u64, low: u64) -> Field {
+    let low_field = low as Field;
+    // We use multiplication instead of bit shifting operations to shift the high bits as bit shift operations are
+    // expensive in circuits.
+    let high_field: Field = (high as Field) * U64_SHIFT_MULTIPLIER;
+    high_field + low_field
+}
+
+pub fn unpack_two_u64s(input: Field) -> (u64, u64) {
+    input.assert_max_bit_size::<128>();
+    let low = (input as u64);
+    // Use division instead of bit shift since bit shifts are expensive in circuits
+    let high = ((input - (low as Field)) / U64_SHIFT_MULTIPLIER) as u64;
+    (high, low)
+}
```

Comment:
I'd not really bother with this being a generic utility - we'll likely never have to do this again. I'd just make this a log encoding specific utility that is rather `fn to_expanded_metadata(log_type, log_metadata)` and `fn from_expanded_metadata`. We'll add more similar fns in the future as log encoding becomes more complicated.

Reply:
Addressed in the one but last commit